code/presentation/mb-client/src/screens/home/masjid-screen.tsx

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable max-lines-per-function */
+import { useNavigation, useRoute } from '@react-navigation/native';
+import React, { useState } from 'react';
+import { Button, StyleSheet, Text, View } from 'react-native';
+
+import { useMasjids } from '@/api/masjid/use-masjids';
+import { getNextNamaz } from '@/api/masjid/util';
+import { masjid } from '@/core/supabase';
+import type { RouteProp } from '@/navigation/types';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#333',
+  },
+  text: {
+    color: 'green',
+    marginBottom: 10,
+    fontSize: 18,
+    fontWeight: 'bold',
+    borderWidth: 1,
+    borderColor: 'green',
+    padding: 10,
+    borderRadius: 5,
+  },
+});
+
+const MasjidScreen = () => {
+  const { data: masjidsWithDistance, isLoading, isError } = useMasjids();
+  const { params } = useRoute<RouteProp<'MasjidScreen'>>();
+  const selectedMasjidId = params.selectedMasjidId;
+  const navigation = useNavigation();
+  const [subscribed, setSubscribed] = useState(false); // State to track subscription status
+
+  if (isLoading) {
+    return (
+      <View>
+        <Text>Loading...</Text>
+      </View>
+    );
+  }
+
+  if (isError || !masjidsWithDistance) {
+    return (
+      <View>
+        <Text>Error occurred while fetching masjid data.</Text>
+      </View>
+    );
+  }
+
+  const selectedMasjid = masjidsWithDistance.find(
+    (masjid) => masjid.masjid.id === selectedMasjidId
+  );
+
+  if (!selectedMasjid) {
+    return (
+      <View>
+        <Text>Selected masjid not found.</Text>
+      </View>
+    );
+  }
+
+  const nextNamaz = getNextNamaz(
+    new Date().toLocaleTimeString(),
+    selectedMasjid.masjid.namaz_timings
+  );
+
+  // Navigation function for subscribing
+  const handleSubscribe = async () => {
+    await masjid.subscribe(selectedMasjidId);
+    setSubscribed(true); // Update subscription status after subscribing
```

Comment:
if we set client state without using the response, how would we know if the request was successful?

if it wasn't successful, client state would show incorrect information

Reply:
Used try catch